kodilla-testing/src/main/java/com/kodilla/testing/forum/statistics/StatisticsCollector.java

```diff
@@ -0,0 +1,69 @@
+package com.kodilla.testing.forum.statistics;
+
+import java.util.List;
+
+public class StatisticsCollector {
+
+    private List<String> usersNames;
+    private int usersQuantity;
+    private int postsQuantity;
+    private int commentsQuantity;
+    private double avgQuantityPostsOnUser;
+    private double avgQuantityCommentsOnUser;
+    private double avgQuantityCommentsOnPosts;
+
+    public int getUsersQuantity() {
+        return usersQuantity;
+    }
+
+    public int getPostsQuantity() {
+        return postsQuantity;
+    }
+
+    public int getCommentsQuantity() {
+        return commentsQuantity;
+    }
+
+    public double getAvgQuantityPostsOnUser() {
+        return avgQuantityPostsOnUser;
+    }
+
+    public double getAvgQuantityCommentsOnUser() {
+        return avgQuantityCommentsOnUser;
+    }
+
+    public double getAvgQuantityCommentsOnPosts() {
+        return avgQuantityCommentsOnPosts;
+    }
+
+    public void calculateAdvStatistics(Statistics statistics) {
+
+        usersNames = statistics.usersNames();
+        usersQuantity = usersNames.size();
+        postsQuantity = statistics.postsCount();
+        commentsQuantity = statistics.commentsCount();
+
+        if (usersQuantity > 0) {
+            avgQuantityPostsOnUser = (double) postsQuantity / usersQuantity;
+            avgQuantityCommentsOnUser = (double) commentsQuantity / usersQuantity;
+        }
+
+        if (postsQuantity > 0)
+            avgQuantityCommentsOnPosts = (double) commentsQuantity / postsQuantity;
+    }
+
+    public double roundTo2(double value) {
+
+        value = ((double)((int)(value *100.0)))/100.0;
+        return value;
+    }
+
+    public String showStatistics() {
+
+        return "There is " + usersQuantity + " users in our forum.\n"
```

Comment:
Zbyt skomplikowana treść metody toString. Zazwyczaj chcemy to mieć w jak najbardziej zwięzły sposób. Np. {users: 10, posts: 100, comments: 40, ... } itp itd. A tak w ogóle, to po co to? Raczej dla klas utilsowych nie piszemy metod toString.